boards/tenstorrent/tt_blackhole/tt_blackhole_fixed_partitions.dtsi

```diff
@@ -14,101 +20,214 @@ partitions {
+		reg = <0x1fb000 DT_SIZE_K(4)>;
+		binary-path =
+			"$BUILD_DIR/smc/generated_board_cfg/$BOARD_REV/flash_info.bin";
+		read-only;
+	};
+
+	/* Remote chip firmware images */
+	ethfw: partition@1fc000 {
+		label = "ethfw";
+		reg = <0x1fc000 DT_SIZE_K(64)>;
+		binary-path = "$BLOBS_DIR/tt_blackhole_erisc.bin";
+		read-only;
+	};
+
+	memfw: partition@20c000 {
+		label = "memfw";
+		reg = <0x20c000 DT_SIZE_K(64)>;
+		binary-path = "$BLOBS_DIR/tt_blackhole_gddr_init.bin";
 		read-only;
 	};
 
-	ethsdfw: partition@47000 {
+	ethsdfw: partition@21c000 {
 		label = "ethsdfw";
-		reg = <0x47000 DT_SIZE_K(20)>;
+		reg = <0x21c000 DT_SIZE_K(64)>;
 		binary-path = "$BLOBS_DIR/tt_blackhole_serdes_eth_fw.bin";
 		read-only;
 	};
 
-	bmfw: partition@4c000 {
-		label = "bmfw";
-		reg = <0x4c000 DT_SIZE_K(68)>;
+	/* Update firmware image for the DMC, as seen by mcuboot */
+	dmfw: partition@22d000 {
+		label = "dmfw";
+		reg = <0x22d000 DT_SIZE_K(452)>;
+	};
+
+	/*
+	 * Partition where image is actually written. Offset by 4KB to
+	 * support swap using move.
+	 */
+	dmfwimg: partition@22e000 {
+		label = "dmfwimg";
+		reg = <0x22e000 DT_SIZE_K(448)>;
 		binary-path = "$BUILD_DIR/dmc/zephyr/zephyr.signed.bin";
-		read-only;
 	};
 
-	flshinfo: partition@59000 {
-		label = "flshinfo";
-		reg = <0x59000 DT_SIZE_K(4)>;
-		binary-path =
-			"$BUILD_DIR/smc/generated_board_cfg/$BOARD_REV/flash_info.bin";
-		read-only;
+	/*
+	 * Image trailer for the DMC firmware update. Indicates to MCUBoot that
+	 * the image is valid and should be booted in test mode.
+	 */
+	dmfw_tail: partition@29d000 {
+		label = "dmfwtail";
+		reg = <0x29d000 DT_SIZE_K(4)>;
+		binary-path = "$BLOBS_DIR/mcuboot_magic_test.bin";
 	};
 
-	boardcfg: partition@5a000 {
-		label = "boardcfg";
-		reg = <0x5a000 DT_SIZE_K(4)>;
-		binary-path = "$BUILD_DIR/smc/generated_board_cfg/$BOARD_REV/read_only.bin";
-		read-only;
+	/*
+	 * This partition is used for the primary firmware image.
+	 * it is loaded by MCUBoot. It is tagged as
+	 * mainimg in tt-boot-fs, but isn't used by the bootrom.
+	 */
+	smc_main_image: partition@29e000 {
+		label = "mainimg";
+		reg = <0x29e000 DT_SIZE_K(512)>;
+		binary-path = "$BUILD_DIR/smc/zephyr/zephyr.signed.bin";
 	};
 
 	/*
-	 * FIXME: The address of this node triggers a warning that we should try to fix:
-	 * Warning (unique_unit_address_if_enabled):
-	 *  /soc/spi@80070000/eeprom@0/partitions/partition@10000000
-	 * duplicate unit-address (also used in node
-	 *   /soc/spi@80070000/eeprom@0/partitions/failover_partition@10000000)
+	 * This partition defines a binary blob that indicates the
+	 * main image should be booted in test mode by mcuboot.
 	 */
-	failover: failover_partition@10000000 {
-		label = "failover";
-		reg = <0x10000000 DT_SIZE_K(76)>;
-		binary-path = "$BUILD_DIR/recovery/zephyr/zephyr.bin";
-		read-only;
+	main_test: partition@31d000{
+		label = "maintail";
+		reg = <0x31d000 DT_SIZE_K(4)>;
+		binary-path = "$BLOBS_DIR/mcuboot_magic_test.bin";
+	};
```

Comment:
I feel like the comments here are also misleading, since the trailer is actually part of the slot0 / slot1 partition.

https://docs.mcuboot.com/design.html#image-trailer

Just a word of caution about over-commenting.  Comments are not code, aren't compiled, and tend to suffer from bitrot. They can't really be seen as a source of truth, and can sometimes be inaccurate, which causes some confusion.